src/application_settings/parametrization.py

```diff
@@ -0,0 +1,172 @@
+"""Defines parameters for application_settings"""
+
+# pylint: disable=duplicate-code
+
+import sys
+from abc import ABC, abstractmethod
+from dataclasses import fields
+from typing import Any, Optional, cast
+
+from attributes_doc import attributes_doc
+from loguru import logger
+from pydantic.dataclasses import dataclass, is_pydantic_dataclass
+
+from application_settings.parameter_kind import ParameterKind, ParameterKindStr
+from application_settings.protocols import ParameterContainerSectionProtocol
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+# This is not an example how to implement a ConfigSection
+# because class ApplicationSettingsContainerSectionBase (re-)implements all methods of ContainerSectionBase.
+# This is done so that this class will meet the ParameterContainerSection protocol
+# yet does not inherit from ContainerSectionBase and therewith prevents circular imports.
+
+
+def log_level(kind: ParameterKind) -> str:
+    """Return the log level taking into consideration whether or not strict_mode is applied"""
+    if kind == ParameterKind.CONFIG:
+        return "WARNING" if ApplicationConfigSection.get().strict_mode else "DEBUG"
```

Comment:
Unexpected that we're returning strings here. Is there not some proper log level enum?

Reply:
hmm seems not in logaru, only integers are possible, so let's keep it like this.

Reply:
Okay.